wnprc_billing/src/org/labkey/wnprc_billing/dataentry/NonAnimalChargesFormType.java

```diff
@@ -0,0 +1,79 @@
+package org.labkey.wnprc_billing.dataentry;
+
+import org.labkey.api.ehr.dataentry.DataEntryFormContext;
+import org.labkey.api.ehr.dataentry.FormSection;
+import org.labkey.api.ehr.dataentry.TaskForm;
+import org.labkey.api.ehr.dataentry.TaskFormSection;
+import org.labkey.api.ehr_billing.security.EHR_BillingAdminPermission;
+import org.labkey.api.module.Module;
+import org.labkey.api.view.template.ClientDependency;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+public class NonAnimalChargesFormType extends TaskForm {
+
+    public NonAnimalChargesFormType(DataEntryFormContext ctx, Module owner)
+    {
+        super(ctx, owner, "NonAnimalCharges", "Enter Charges without Animal Ids", "Billing", Arrays.asList(
+                new TaskFormSection(),
+                new NonAnimalChargesFormSection()
+        ));
+
+        addClientDependency(ClientDependency.fromPath("ehr_billing/buttons/financeButtons.js"));
+        addClientDependency(ClientDependency.fromPath("ehr_billing/DataEntryUtils.js"));
+        addClientDependency(ClientDependency.fromPath("ehr/DataEntryUtils.js"));
+        addClientDependency(ClientDependency.fromPath("ehr_billing/form/field/EHRBillingRowObserverEntryField.js"));
+        addClientDependency(ClientDependency.fromPath("wnprc_billing/model/sources/NonAnimalCharges.js"));
+        addClientDependency(ClientDependency.fromPath("wnprc_billing/form/field/ChargeItemField.js"));
+        addClientDependency(ClientDependency.fromPath("wnprc_billing/form/field/MiscChargesDateField.js"));
+        addClientDependency(ClientDependency.fromPath("wnprc_billing/form/field/ChargeTypeField.js"));
+        addClientDependency(ClientDependency.fromPath("wnprc_billing/form/field/MiscChargesDebitAcctEntryField.js"));
```

Comment:
Since both this form type and the with animal id one share most of these dependencies, would it make sense to factor out an abstract class they can both extend from and have that parent abstract class add all the shared dependencies?

Reply:
I assume that parent abstract class might also be able to have most of the code below this for this form type java class.

Reply:
Good call! Abstract form type is now created to share common dependencies and methods.